website/docs: document npm install-script blocking (#22461)
goauthentik/authentik: website/docs/developer-docs/setup/_npm-install-scripts-admonition.mdx

:::info NPM install scripts are disabled by default

> 

Reviewer: ```suggestion
```

> The repository's `.npmrc` sets `ignore-scripts=true` so `preinstall`/`install`/`postinstall` lifecycle scripts do not run during `npm ci`. This neutralizes a dominant NPM supply-chain attack pattern at the cost of skipping a few legitimate native-binary unpacks.

Reviewer: ```suggestion
The repository's NPM runtime configuration file (`.npmrc`) sets `ignore-scripts=true`. This means that `preinstall`/`install`/`postinstall` lifecycle scripts do not run during `npm ci`. This neutralizes a dominant NPM supply-chain attack pattern at the cost of skipping a few legitimate native-binary unpacks.
```

> If the watch build fails because a package needs its install script (commonly `esbuild`, `chromedriver`, `tree-sitter`, or `tree-sitter-json`), rebuild only that package:

Reviewer: ```suggestion
If the watch build fails because a package needs its install script (commonly `esbuild`, `chromedriver`, `tree-sitter`, or `tree-sitter-json`), rebuild only that package, for example:
```

```shell
npm rebuild --foreground-scripts esbuild chromedriver tree-sitter tree-sitter-json
```

> **Do not** edit `.npmrc` to flip `ignore-scripts` off — that re-introduces the risk repo-wide.

Reviewer: ```suggestion
**Do not** edit `.npmrc` to flip `ignore-scripts` off — that re-introduces the risk repository-wide.
```

> **Do not** edit `.npmrc` to flip `ignore-scripts` off — that re-introduces the risk repo-wide.

Reviewer 2: ehhhh I'd say people can do what they please on their local machines (if not corp owned, but considering as this is contribution docs, probably just extern) We have the file for a reason and I don't really see the need to outline the fact to not modify this one over others. Anyways, if any npm files are modified we'll see it when a pr is opened

Reviewer: I think it's fine to mention it explicitly just in case some LLM tells them to do it as a quick fix. They can always ignore the note.

> 

Reviewer: ```suggestion
```
